romancal/photom/photom.py

```diff
@@ -0,0 +1,169 @@
+import logging
+import warnings
+
+import numpy as np
+from astropy import units as u
+
+log = logging.getLogger(__name__)
+log.setLevel(logging.DEBUG)
+
+# Conversion factor from MJy/sr to uJy/arcsec^2
+MJSR_TO_UJA2 = (u.megajansky / u.steradian).to(u.microjansky / u.arcsecond / u.arcsecond)
+
+# Conversion factor from square arcseconds to steradians
+A2_TO_SR = (np.pi / (180. * 3600.))**2
+
+
+def photom_io(input_model, photom_metadata):
+    """
+    Short Summary
+    -------------
+    Combine photometric scalar conversion factors and add to the science metadata.
+
+    Parameters
+    ----------
+    input_model : Roman level 2 image datamodel
+        input roman datamodel
+
+    photom_metadata : list
+        Set of matched photom meta data keys
+
+    Returns
+    -------
+
+    """
+    # Get the scalar conversion factor.
+    conversion = photom_metadata['photmjsr']  # unit is MJy / sr
+
+    # Store the conversion factor in the meta data
+    log.info(f'photmjsr value: {conversion:.6g}')
+    input_model.meta.photometry.conversion_megajanskys = conversion
+    input_model.meta.photometry.conversion_microjanskys = conversion.value * MJSR_TO_UJA2 *\
+                                                          u.microjansky / (u.arcsecond**2)
+
+    # Get the scalar conversion uncertainty factor
+    uncertainty_conv = photom_metadata['uncertainty']
+
+    # Store the uncertainty conversion factor in the meta data
+    log.info(f'uncertainty value: {conversion:.6g}')
+    input_model.meta.photometry.conversion_megajanskys_uncertainty = uncertainty_conv
+    input_model.meta.photometry.conversion_microjanskys_uncertainty = uncertainty_conv.value * \
+                                                  MJSR_TO_UJA2 * u.microjansky / (u.arcsecond**2)
+
+    # Return updated input model
+    return input_model
+
+
+def find_photom_parameters(photom_table, match_fields):
+    """
+    Find metadata keyword object matching fields.
+
+    Parameters
+    ----------
+    photom_table : dict
+        "Optical Element":
+            {"photmjsr": value * u.MJ / u.sr,
+             "uncertainty": value* u.MJ / u.sr,
+             "pixelareasr": value * u.sr}
+    match_fields : dict
+        {field_name: value} pair to use as a matching criteria.
+
+    Raises
+    ------
+    Warning
+        When a field name is not in the table.
+    MatchFitsTableRowError
```

Comment:
I don't see this error being raised, did I miss it?
Also I don't think this is from a FITS table for Roman

Reply:
Yes, that was a straggler that needed to be deleted. Both cases raise a warning - tweaked for that.